eigsep_motor_control/encoder.py

```diff
@@ -27,70 +27,76 @@ def get_encoder(self, motor):
 class Potentiometer:
 
     NBITS = 16  # ADC number of bits
-
-    # serial connection constants (BAUDRATE defined in main.py)
-    PORT = "/dev/ttyACM0"
+    PORT = "/dev/ttyACM0"  # serial connection constants
 
     def __init__(self):
-        """
-        Class for reading voltages from the potentiometers.
-        """
-        self.ser = serial.Serial(port=self.PORT, baudrate=BAUDRATE)
-        self.ser.reset_input_buffer()
+        while True:
+            try:
+                self.ser = serial.Serial(port=self.PORT, baudrate=BAUDRATE)
+                self.ser.reset_input_buffer()
+                break
+            except serial.SerialException as e:
+                logging.error(f"Failed to open serial port {self.PORT}: {e}. Trying again.")
+                # Additional error handling or recovery actions can be placed here
+                continue #try again
 
-        # voltage range of the pots
         self.VOLT_RANGE = {"az": (0.7, 1.7), "alt": (0.7, 1.7)}
 
-    def bit2volt(self, analog_value):
+    def bit2volt(self, analog_value, motor):
+        vmax = self.VOLT_RANGE[motor][1]
         res = 2**self.NBITS - 1
-        ratio = self.VMAX / res
+        ratio = vmax / res
         return ratio * analog_value
 
     def read_analog(self):
-        """
-        Read the analog values of the pots.
-
-        Returns
-        -------
-        data : np.ndarray
-            The analog values of the pots averaged over INT_LEN
-            measurements. The first value is associated with the azimuth
-            pot, the second value is the altitude pot.
-
-        """
-        data = self.ser.readline().decode("utf-8").strip()
-        data = [int(d) for d in data.split()]
-        return np.array(data) / INT_LEN
-
+        while True:
```

Comment:
Not sure the effects of this while loop downstream (e.g. in the monitor function) or other scripts. My preference might be to just return None in the Except block and then handle it in other scripts